modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/ReflectiveMethodComponentExecutor.java

```diff
@@ -131,16 +138,14 @@ public void setMuleContext(MuleContext context) {
 
   @Override
   public Function<ExecutionContext<M>, Map<String, Object>> createArgumentResolver(M operationModel) {
-    return ec -> withContextClassLoader(extensionClassLoader,
-                                        () -> {
-                                          final Object[] resolved =
-                                              getParameterValues(ec, method.getParameterTypes());
-
-                                          final Map<String, Object> resolvedParams = forSize(parameterCount);
-                                          for (int i = 0; i < parameterCount; ++i) {
-                                            resolvedParams.put(method.getParameters()[i].getName(), resolved[i]);
-                                          }
-                                          return resolvedParams;
-                                        });
+    return ec -> withContextClassLoader(extensionClassLoader, () -> {
+      final Object[] resolved = argumentResolverDelegate.resolveDeferred(ec, methodParameterTypes);
```

Comment:
resolveDeferred returns a Supplier<Object>[] , did you just use Object[] here because you did not care that it was an array of Suppliers?